helm/resource_release.go

```diff
@@ -566,7 +566,18 @@ func resourceReleaseCreate(ctx context.Context, d *schema.ResourceData, meta int
 			return diag.FromErr(err)
 		}
 
-		return nil
+		warnAndError := diag.Diagnostics{
+			{
+				Severity: diag.Warning,
+				Summary:  fmt.Sprintf("Helm created release \"%s\" but it is in a failed state. Persisting it for troubleshooting purposes.", client.ReleaseName),
```

Comment:
you can use `%q` instead of `\"%s\"` here

Reply:
We could also change the wording here to be a bit more actionable – persist makes it sound like we've persisted the Terraform state of the resource, which in this case we actually haven't. In this case we probably want to tell the user that they'll need to do two things before they can run apply again:

1. Solve/fix the error
2. Delete the failed release

How about:

```
Helm release %q was created but has a failed status. Use the `helm` command to investigate the error then delete the release before running Terraform again.
```

Reply:
> One downside of this approach is that now you have to use `helm` to delete the release before you can run apply again – `terraform destroy` wont remove it and you'll get an "already exists" error.

So it looks like the release in TF gets tainted, so the next apply forces a replacement. This seems to work with my test chart. I remove the namespace definition that doesn't exist, TF deletes and re-creates the Helm release, and Helm reports the release like so:
```
NAME: should-fail
LAST DEPLOYED: Tue Apr 20 10:27:21 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
```
So I think this is actually a pretty good workflow.

Reply:
Nice! That's great then.